### PR TITLE
Add luminosity channels

### PR DESCRIPTION
**Pull request by Zaharid**
This adds lumi channels corresponding to the main production channels for W and Z bosons.

### CODE REVIEW

**Comment by Zaharid** [comment]
Example https://vp.nnpdf.science/aAiBE9KfTYuDLv0jb8iGQw==/

File: validphys2/src/validphys/gridvalues.py [hunk 1]
```diff
@@ -155,11 +166,24 @@ def evaluate_luminosity(
...
+            _parton_pair_lumi_inner(pdf_set=pdf_set, n=n, mx=mx, x1=x1, x2=x2, i=u, j=ubar) +
+            _parton_pair_lumi_inner(pdf_set=pdf_set, n=n, mx=mx, x1=x1, x2=x2, i=d, j=dbar)
+        )
+    elif channel == 'wlum':
+        u, dbar = 2, -1
+        d, ubar = 1, -2
+        res = (
+            _parton_pair_lumi_inner(pdf_set=pdf_set, n=n, mx=mx, x1=x1, x2=x2, i=u, j=dbar) +
+            _parton_pair_lumi_inner(pdf_set=pdf_set, n=n, mx=mx, x1=x1, x2=x2, i=d, j=ubar)
+        )
```
Comment (scarlehoff) [inline validphys2/src/validphys/gridvalues.py:182]: Do you want `u ub` x `d db` only or all  `u-like ub-like`? I think it makes more sense for a `zlum` to have all generations.

**Comment by scarlehoff** [comment]
Wouldn't it be better to have a parser so that one can use any combination? We have all the information necessary to understand things like: 
```
lumi_channel: (u ubar, d dbar)
```
available https://github.com/NNPDF/nnpdf/blob/7eb0c4eb67604a7d0b80f9a8ac075aab994904cc/validphys2/src/validphys/pdfbases.py#L533

File: validphys2/src/validphys/gridvalues.py (hunk 1 above)
Comment (Zaharid) [reply validphys2/src/validphys/gridvalues.py:182]: I want these lumis now

**Comment by Zaharid** [comment]
Baring the very flaky tests, I'd like to merge this as is.

File: validphys2/examples/pdf_lumi_plots.yaml [hunk 2]
```diff
@@ -13,7 +13,8 @@ pdf: {id: "NNPDF40_nlo_as_01180", label: "4.0 NLO"}
 sqrts: 13000 # GeV
 
 lumi_channel: "gg" # one of [gg, gq, qqbar, qq, ddbar, uubar, ssbar,
-                   #         ccbar, bbbar, dubar, udbar, scbar, csbar, pp, gp]
+                   #         ccbar, bbbar, dubar, udbar, scbar, csbar, pp, gp,
+                   #         wlum, zlum]
```
Comment (scarlehoff) [inline validphys2/examples/pdf_lumi_plots.yaml:17]:
```suggestion
                   #         wlum1, zlum1]
```

File: validphys2/src/validphys/gridvalues.py [hunk 3]
```diff
@@ -32,6 +32,8 @@
     'csbar': r'c\bar{s}',
     'pp': r'\gamma\gamma',
     'gp': r'g\gamma',
+    'zlum': r'u\bar{u} + d\bar{d}',
+    'wlum': r'u\bar{d} + d\bar{u}',
```
Comment (scarlehoff) [inline validphys2/src/validphys/gridvalues.py:36]:
```suggestion
    'zlum1': r'u\bar{u} + d\bar{d}',
    'wlum1': r'u\bar{d} + d\bar{u}',
```

File: validphys2/src/validphys/gridvalues.py [hunk 4]
```diff
@@ -155,11 +166,24 @@ def evaluate_luminosity(
 
         # as in the second of Eq.(4) in arXiv:1607.01831
         res = sum(a*b for a,b in itertools.product(r1,r2))
+    elif channel == 'zlum':
```
Comment (scarlehoff) [inline validphys2/src/validphys/gridvalues.py:169]:
```suggestion
    elif channel == 'zlum1':
```

File: validphys2/src/validphys/gridvalues.py [hunk 5]
```diff
@@ -155,11 +166,24 @@ def evaluate_luminosity(
...
+        u, ubar = 2, -2
+        d, dbar = -1, 1
+        res = (
+            _parton_pair_lumi_inner(pdf_set=pdf_set, n=n, mx=mx, x1=x1, x2=x2, i=u, j=ubar) +
+            _parton_pair_lumi_inner(pdf_set=pdf_set, n=n, mx=mx, x1=x1, x2=x2, i=d, j=dbar)
+        )
+    elif channel == 'wlum':
```
Comment (scarlehoff) [inline validphys2/src/validphys/gridvalues.py:176]:
```suggestion
    elif channel == 'wlum1':
```

File: validphys2/src/validphys/gridvalues.py (hunk 1 above)
Comment (scarlehoff) [reply validphys2/src/validphys/gridvalues.py:182]: I wanted just to make sure it was not an oversight.

**Review by scarlehoff — APPROVED** [review]
I've added the comments with the change `Xlum` -> `Xlum1` to signal it is only first generation quarks because I have the feeling this is the kind of thing that will come back to bite it in the future "wait, why does the Z luminosity remain unchanged even if the charm has changed a lot"

But not a worthy hill to die on and I'm happy also with being able to say "I told you so" when it eventually happens so feel free to merge.

File: validphys2/src/validphys/gridvalues.py (hunk 1 above)
Comment (Zaharid) [reply validphys2/src/validphys/gridvalues.py:182]: Fair enough.

**Pull request merged**